### DISCUSSION TITLE
How can I access the return value of a fixture defined via @pytest.mark.usefixtures

### DISCUSSION THREAD

**Discussion by FelixKJose** [pytest-dev/pytest#8816] [Q&A] [answered]
I am not able to access the return value of a fixture when it is defined via @pytest.mark.usefixtures.

Eg:
**conftest.py:**
```
@pytest.fixture
def test_envs():
    return {"data": ["dev", "staging"]}
```
Following doesn't work:
**test_example_fixture.py**

```
@pytest.mark.usefixtures("test_envs")
class Test:
    def test_fixtures(self):
        print("Fixture values:", self.pytestmark[0]["data"])
```

```
def test_fixtures_func(test_envs):
        print("Fixture values:", test_envs["data"])
```
        
worked. So you cannot use "@pytest.mark.usefixtures" to add fixtures for all tests in a Test class or module and access the defined fixtures return value in each test?

**Comment by RonnyPfannschmidt** [comment, marked as answer]
if you want to use the value of a fixture, take it as a function argument

pytest will not under any circumstance monkeypatch fixtures into other places
